scene/resources/particle_process_material.cpp

```diff
@@ -750,57 +759,55 @@ void ParticleProcessMaterial::_update_shader() {
 		code += "	return normalize(spread_direction);\n";
 	}
-	code += "}\n";
+	code += "}\n\n";
 
-	code += "vec3 process_radial_displacement(DynamicsParameters param, float lifetime, inout uint alt_seed, mat4 transform, mat4 emission_transform, float delta){\n";
+	code += "vec3 process_radial_displacement(DynamicsParameters param, float lifetime, inout uint alt_seed, mat4 transform, mat4 emission_transform, float delta) {\n";
 	code += "	vec3 radial_displacement = vec3(0.0);\n";
-	code += "	if (delta < 0.001){\n";
+	code += "	if (delta < 0.001) {\n";
 	code += "		return radial_displacement;\n";
 	code += "	}\n";
 	code += "	float radial_displacement_multiplier = 1.0;\n";
 	if (tex_parameters[PARAM_RADIAL_VELOCITY].is_valid()) {
 		code += "	radial_displacement_multiplier = texture(radial_velocity_curve, vec2(lifetime)).r;\n";
 	}
 	code += "	vec3 global_pivot = (emission_transform * vec4(velocity_pivot, 1.0)).xyz;\n";
-	code += "	if(length(transform[3].xyz - global_pivot) > 0.01){\n";
+	code += "	if (length(transform[3].xyz - global_pivot) > 0.01) {\n";
 	code += "		radial_displacement = normalize(transform[3].xyz - global_pivot) * radial_displacement_multiplier * param.radial_velocity;\n";
-	code += "	}else{radial_displacement = get_random_direction_from_spread(alt_seed, 360.0)* param.radial_velocity;} \n";
-	code += "	if (radial_displacement_multiplier * param.radial_velocity < 0.0){\n // Prevent inwards velocity to flicker once the point is reached.";
-	code += "		if (length(radial_displacement) > 0.01){\n";
-	code += "		radial_displacement = normalize(radial_displacement) * min(abs((radial_displacement_multiplier * param.radial_velocity)), length(transform[3].xyz - global_pivot) / delta);\n";
-	code += "		}\n";
-	code += "\n";
+	code += "	} else {\n";
+	code += "		radial_displacement = get_random_direction_from_spread(alt_seed, 360.0) * param.radial_velocity;\n";
+	code += "	}\n";
+	code += "	if (radial_displacement_multiplier * param.radial_velocity < 0.0) {\n";
+	code += "		// Prevent inwards velocity to flicker once the point is reached.\n";
+	code += "		radial_displacement = normalize(radial_displacement) * min(abs(radial_displacement_multiplier * param.radial_velocity), length(transform[3].xyz - global_pivot) / delta);\n";
+	code += "	}\n";
 	code += "	return radial_displacement;\n";
-	code += "}\n";
+	code += "}\n\n";
+
 	if (tex_parameters[PARAM_DIRECTIONAL_VELOCITY].is_valid()) {
-		code += "vec3 process_directional_displacement(DynamicsParameters param, float lifetime_percent,mat4 transform, mat4 emission_transform){\n";
-		code += "	vec3 displacement = vec3(0.);\n";
+		code += "vec3 process_directional_displacement(DynamicsParameters param, float lifetime_percent, mat4 transform, mat4 emission_transform) {\n";
+		code += "	vec3 displacement = texture(directional_velocity_curve, vec2(lifetime_percent)).xyz * param.directional_velocity;\n";
```

Comment:
I just moved the line present in both branches out of the `if` check, removing the need for the `else`.